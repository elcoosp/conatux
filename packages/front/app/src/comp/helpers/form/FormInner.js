import React from 'react'

import P from 'prop-types'
import { Form as UiForm, Submit } from 'ui/form'

const FormInner = ({ children, isSubmitting, handleReset, handleSubmit }) => (
  <UiForm onReset={handleReset} onSubmit={handleSubmit}>
    {children}
    <Submit disabled={isSubmitting} />
  </UiForm>
)

FormInner.propTypes = {
  children: P.node.isRequired,
  isSubmitting: P.bool.isRequired,
  handleReset: P.func.isRequired,
  handleSubmit: P.func.isRequired
}

export default FormInner
