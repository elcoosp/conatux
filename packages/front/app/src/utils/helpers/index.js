export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
