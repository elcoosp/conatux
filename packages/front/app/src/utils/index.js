export { default as testing } from './testing'
export { default as linkify } from './linkify'
