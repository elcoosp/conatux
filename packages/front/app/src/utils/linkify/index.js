import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
