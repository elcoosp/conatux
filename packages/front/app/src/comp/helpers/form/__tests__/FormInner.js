import { desc, getFakeProps } from '@:/utils/testing'
import FormInner from '../FormInner'
const [props] = getFakeProps(__dirname, '../FormInner')('FormInner')

desc.compWithSnap({ FormInner }, props, () => {})
