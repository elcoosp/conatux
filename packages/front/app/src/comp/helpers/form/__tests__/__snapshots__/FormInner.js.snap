// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInner /> should match snapshot 1`] = `
<styled.form
  onReset={[Function]}
  onSubmit={[Function]}
>
  children
  <Submit
    disabled={true}
  />
</styled.form>
`;
